### ISSUE TITLE
Feature request: parser for `dpkg -l` and `rpm -qa`

### ISSUE THREAD

**Issue by tst2005** [kellyjonbrazil/jc#112]
I think it should be cool to have parser for installed packages.
I don't know if it is a hard work or not.

about: dpkg -l you could probably keep only line starting with 'ii ' and 'rc '

Regards,

**Comment by kellyjonbrazil** [comment]
Hi! Yes, these would be good to add - let me see if I can fit them in in the next couple weeks. Table and list output like this is pretty easy to parse, typically.

**Comment by kellyjonbrazil** [comment]
Hi there - I have a working version of the `dpkg -l` parser. You can try it out by copy/pasting the code from the link below to a file called`~/.local/share/jc/jcparsers/dpkg_l.py`:

https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/dpkg_l.py

let me know if you run into any issues.

Thanks!

**Comment by kellyjonbrazil** [comment]
Could you give me an example of the `rpm -qa` output you would like parsed? on my Centos 7.7 VM I'm only seeing a simple list of filenames. I'm not sure there's much value in turning that into JSON. I suppose I could output an array of filenames, but that would be different than all of the other parsers.

In fact, I think `jq` do this natively by slurping the list:
```
rpm -qa | jq -R | jq -s
```

**Comment by tst2005** [comment]
It is true, the `rpq -qa` is a simple list.
I dreams to have a splited name, version and arch fields but it seems hard to parse because is it seems not really strictly named.

```
jq -R . | jq '
. as $rawname |
split(".") as $s|
($s|last) as $arch |
($s|.[:-1]|join(".")) as $woarch|
($woarch|capture("(?<name>.*)-(?<version>[0-9]\\.[0-9]+.*)")) as $x|
{name:$x.name, version:$x.version, arch:$arch}
'
```

Input sample:
```
libjose-10-2.el8.x86_64
graphite2-1.3.10-10.el8.x86_64
mozjs60-60.9.0-4.el8.x86_64
python3-gpg-1.10.0-6.el8.0.1.x86_64
glibc-langpack-en-2.28-101.el8.x86_64
bc-1.07.1-5.el8.x86_64
```

Output sample:
```
{
  "name": "graphite2",
  "version": "1.3.10-10.el8",
  "arch": "x86_64"
}
{
  "name": "python3-gpg",
  "version": "1.10.0-6.el8.0.1",
  "arch": "x86_64"
}
{
  "name": "glibc-langpack-en",
  "version": "2.28-101.el8",
  "arch": "x86_64"
}
{
  "name": "bc",
  "version": "1.07.1-5.el8",
  "arch": "x86_64"
}
```

- you could do the same with jc ?

**Comment by kellyjonbrazil** [comment]
Here is a working version of the `rpm -qai` parser:
https://github.com/kellyjonbrazil/jc/blob/dev/jc/parsers/rpm_qai.py

Let me know if that works for you. I'm planning on adding it to v1.15.0

**Comment by tst2005** [comment]
Thanks, I didn't think about that way.
It seems very nice!

**Comment by kellyjonbrazil** [comment]
v1.15.0 is now available via pip, packages, and binaries.
